### ISSUE TITLE
Proposal on passing configuration

### ISSUE THREAD

**Issue by slint** [data-futures/mirador3-dist-builder#1]
To avoid computing manifest, canvas, and annotation URLs by manipulating `window.location` attributes, there should be explicit ways to pass this information to the Mirador viewer config:

- The existing `m` and `c` search params are a good existing method 
  - This could be extended with an `a` parameter for passing the annotation URL
- Relying on HTML `data-` attributes, on the root element with the `id` under which Mirador is mounted, e.g.:

```html
<div 
  id="m3-dist" 
  data-manifest="https://zenodo.org/api/iiif/record:1234/manifest"
  data-canvas="https://zenodo.org/api/iiif/record:1234/canvas/page-01.png"
  data-annotations="https://zenodo.org/api/records/1234/media-files/annotations.json/content"
</div>
```

- Another alternative would be to export JS function for wrapping the initialization, to also for passing config overrides, e.g.:

```js
function initMirador({ id, manifest, canvas, annotations, config }) {
  const config = {
    id: id,
    windows: [{
      manifestId: manifest,
      canvasId: canvas,
    }],
    // ...default config...
    
    // allow also to override default config
    ...(config || {}),
  };

  Mirador.viewer(config);
}
```

**Comment by dfdan** [comment]
Hi Alex,

Thanks for this - the window.location kludge should definietly be considered a placeholder. 

I think the data-attributes is probably the tidiest soloution, although I'd like to keep the URL params for some legacy things (despite the risk of the URL getting too long). But in my view your proposed data-attribute solution should be the "primary" configuration method, as we can use it flexibly to pass additional config variables (and in some cases maybe even a whole Mirador config). 

Look forward to talking more about this later!
(pinging @peter-cornwell)

**Comment by dfdan** [comment]
closed by #3
